### PR TITLE
Add super stream producer

### PR DESCRIPTION
**Pull request by Gsantomaggio**
## SuperStreamProducer
----

New struct to create the super stream producer:

The main features are:

- `RoutingStrategy` [interface](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/super_stream_3/pkg/stream/super_stream_producer.go#L12-L28)  
- `HashRoutingStrategy` [struct](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/super_stream_3/pkg/stream/super_stream_producer.go#L46)
- `KeyRoutingStrategy` [struct](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/super_stream_3/pkg/stream/super_stream_producer.go#L61)

- `SuperStreamProducer` [struct](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/super_stream_3/pkg/stream/super_stream_producer.go#L166C6-L166C25)
- `PartitionContext` [struct](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/super_stream_3/pkg/stream/super_stream_producer.go#L162C6-L162C22) 

Example:

```golang
      superStreamProducer, err := env.NewSuperStreamProducer(superStreamName,
		stream.NewSuperStreamProducerOptions(
			stream.NewHashRoutingMurmurStrategy(func(message message.StreamMessage) string {
				return message.GetApplicationProperties()["myKey"].(string)
			})).SetClientProvidedName("my-superStreamProducer"))
	CheckErr(err)
```

[Here](https://github.com/rabbitmq/rabbitmq-stream-go-client/blob/super_stream_3/examples/super_stream/super_stream.go) the full code example

### CODE REVIEW

File: pkg/test_helper/http_utils.go [hunk 1]
```diff
@@ -21,19 +21,19 @@ type connection struct {
...
+//		return 0, err
+//	}
+//
+//	var data queue
+//	err = json.Unmarshal([]byte(bodyString), &data)
+//	if err != nil {
+//		return 0, err
+//	}
+//	return data.Messages, nil
+//}
```
Comment (DanielePalaia) [inline pkg/test_helper/http_utils.go:36]: Same as before

File: examples/super_stream/super_stream.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+			break
+		default:
+			CheckErr(err)
+		}
+		time.Sleep(200 * time.Millisecond)
+	}
+
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println("Press enter to close the producer")
+	_, _ = reader.ReadString('\n')
```
Comment (DanielePalaia) [inline examples/super_stream/super_stream.go:142]: Here we don't really need any returned value... can we just reader.ReadString?

File: examples/super_stream/super_stream.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+	err = superStreamProducer.Close()
+	CheckErr(err)
+	fmt.Printf("Producer closed. Total confirmed: %d, total failed: %d total messages: %d\n",
+		confirmed, failed, confirmed+failed)
+
+	fmt.Println("Press enter to exit")
+	_, _ = reader.ReadString('\n')
```
Comment (DanielePalaia) [inline examples/super_stream/super_stream.go:149]: Same a as before

File: pkg/test_helper/http_utils.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package ha
+package test_helper
```
Comment (DanielePalaia) [inline pkg/test_helper/http_utils.go:1]: the name of the package with the underscore may not be good in GO

**Comment by Gsantomaggio** [comment]
Will merge after pairing with @DanielePalaia

**Pull request merged**